apps/condo/domains/common/components/Comments/index.tsx

```diff
@@ -53,106 +71,329 @@ const Head = styled.div`
+const NewCommentIndicator = styled.div`
+    position: relative;
+    top: 5px;
+    width: 4px; 
+    height: 4px; 
+    border-radius: 100px; 
+    background-color: ${colors.red[5]};
+`
+
+const CommentsTabPaneLabel = ({ label, commentsCount, newCommentsIndicator }) => (
+    <>
+        <Typography.Text>
+            {label}
+        </Typography.Text>
+        <Typography.Text style={COMMENTS_COUNT_STYLES}>
+            {commentsCount}
+        </Typography.Text>
+        {
+            newCommentsIndicator && (
+                <NewCommentIndicator title={''} />
+            )
+        }
+    </>
+)
+
 interface ICommentsListProps {
+    ticket: ITicketUIState,
     comments: TComment[],
-    createAction?: (formValues) => Promise<void>,
+    createAction?: (formValues) => Promise<TComment>,
+    updateAction?: (attrs, obj: TComment) => Promise<TComment>
     // Place for abilities check. If action of given type is not returned, appropriate button will not be displayed
     actionsFor: (comment: TComment) => ActionsForComment,
     canCreateComments: boolean,
+    refetchComments,
+    FileModel: Module,
+    fileModelRelationField: string,
 }
 
 const Comments: React.FC<ICommentsListProps> = ({
+    ticket,
     comments,
     createAction,
+    updateAction,
+    refetchComments,
     canCreateComments,
     actionsFor,
+    FileModel,
+    fileModelRelationField,
 }) => {
     const intl = useIntl()
     const TitleMessage = intl.formatMessage({ id: 'Comments.title' })
-    const PromptTitleMessage = intl.formatMessage({ id: 'Comments.prompt.title' })
-    const PromptDescriptionMessage = intl.formatMessage({ id: 'Comments.prompt.description' })
-    const ListDescriptionMessage = intl.formatMessage({ id: 'Comments.list.description' })
     const CannotCreateCommentsMessage = intl.formatMessage({ id: 'Comments.cannotCreateComments' })
+    const InternalCommentsMessage = intl.formatMessage({ id: 'Comments.tab.organization' })
+    const PromptInternalCommentsTitleMessage = intl.formatMessage({ id: 'Comments.tab.organization.prompt.title' })
+    const PromptInternalCommentsDescriptionMessage = intl.formatMessage({ id: 'Comments.tab.organization.prompt.description' })
+    const ResidentCommentsMessage = intl.formatMessage({ id: 'Comments.tab.resident' })
+    const PromptResidentCommentsTitleMessage = intl.formatMessage({ id: 'Comments.tab.resident.prompt.title' })
+    const PromptResidentCommentsDescriptionMessage = intl.formatMessage({ id: 'Comments.tab.resident.prompt.description' })
 
-    const bodyRef = useRef(null)
+    const { user } = useAuth()
     const { isSmall } = useLayoutContext()
+    const [commentType, setCommentType] = useState(ORGANIZATION_COMMENT_TYPE)
+    const [editableComment, setEditableComment] = useState<TComment>()
+    const [sending, setSending] = useState(false)
+
+    const action = useCallback(async (values, syncModifiedFiles) => {
```

Comment:
`action` is not an appropriate name for callback function, please use verb prefix and some short description like `handleCommentAction`